instrumentation/othttp/handler_test.go

```diff
@@ -20,28 +20,52 @@ import (
 	"net/http/httptest"
 	"testing"
 
+	"github.com/stretchr/testify/assert"
+
+	"go.opentelemetry.io/otel/api/kv"
+	"go.opentelemetry.io/otel/api/standard"
+	mockmeter "go.opentelemetry.io/otel/internal/metric"
 	mocktrace "go.opentelemetry.io/otel/internal/trace"
 )
 
+func assertMetricLabels(t *testing.T, expectedLabels []kv.KeyValue, measurementBatches []mockmeter.Batch) {
+	for _, batch := range measurementBatches {
+		assert.ElementsMatch(t, expectedLabels, batch.Labels)
+	}
+}
+
 func TestHandlerBasics(t *testing.T) {
 	rr := httptest.NewRecorder()
 
 	var id uint64
 	tracer := mocktrace.MockTracer{StartSpanID: &id}
+	meterimpl, meter := mockmeter.NewMeter()
+
+	operation := "test_handler"
 
 	h := NewHandler(
 		http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 			if _, err := io.WriteString(w, "hello world"); err != nil {
 				t.Fatal(err)
 			}
-		}), "test_handler",
-		WithTracer(&tracer))
+		}), operation,
+		WithTracer(&tracer),
+		WithMeter(meter),
+	)
 
 	r, err := http.NewRequest(http.MethodGet, "http://localhost/", nil)
 	if err != nil {
 		t.Fatal(err)
 	}
 	h.ServeHTTP(rr, r)
+
+	if len(meterimpl.MeasurementBatches) == 0 {
+		t.Fatalf("got 0 recorded measurements, expected 1 or more")
+	}
+
+	labelsToVerify := standard.HTTPServerMetricAttributesFromHTTPRequest(operation, r)
```

Comment:
Can you construct the verification set without relying on the same function used by the unit under test?

Reply:
Yep, makes sense if someone tries to add some high cardinality labels to this function. will do

Reply:
I don't think this issue was resolved, but the test helpers for writing this test are in a bad shape. There is a helper used internally to the metrics SDK (`sdk/metric/processor/test.Output`) that produces reasonably nice tests, but the framework in internal/metric/mock.go needs more work (note that it's used in only two tests.)

I suggest that we merge this PR. I will file an issue about improving the test framework.

Reply:
#872

Reply:
would like to take it after this PR merged

Reply:
> would like to take it after this PR merged

Sounds good :+1: . I think the idea is to track the work with #872.